core/templates/pages/contributor-dashboard-page/opportunities-list/opportunities-list.component.ts

```diff
@@ -144,17 +138,17 @@ export class OpportunitiesListComponent {
           this.opportunities = this.opportunities.concat(opportunitiesDicts);
           this.visibleOpportunities = this.opportunities.slice(
             startIndex, endIndex);
-          this.userIsOnLastPage = this.calculateUserIsOnLastPage(
-            this.opportunities,
-            this.OPPORTUNITIES_PAGE_SIZE,
-            pageNumber,
-            this.more);
           this.loadingOpportunityData = false;
         });
     } else {
       this.visibleOpportunities = this.opportunities.slice(
         startIndex, endIndex);
     }
+    this.userIsOnLastPage = this.calculateUserIsOnLastPage(
+      this.opportunities,
+      this.OPPORTUNITIES_PAGE_SIZE,
+      pageNumber,
+      this.more);
```

Comment:
Wouldn't you have a race condition here if `this.more` is being updated in the then() block above?

Reply:
Sorry, where did you add it? I don't see it.